### PR TITLE
Refactored library with generated structs and improved error handling

### PR DESCRIPTION
**Pull request by lornajane**
This changes ... basically everything in the library! The main goals:
 - a more sane and flexible user interface so you can configure auth and then per-product operations
 - autogenerated OAS behind the scenes to ensure accuracy and improve maintainability
 - clear docs, tests and examples

This prototype will cover (checkboxes for what's already available):
[x] SMS
[x] Numbers
[x] Verify
[x] JWT features
[x] Number Insight
[x] Voice
[x] Applications

### CODE REVIEW

**Comment by lornajane** [comment]
Note that everything in `sms/` is generated code - the idea is that we can replace this if the spec changes

**Comment by lornajane** [comment]
I just pushed some changes to the error handling in the Verify section - check the README for examples and let me know what you think? ( @superdiana thanks for the moral support on this!)

**Comment by seopei** [comment]
any updates when this one will get merged and released?

**Comment by lornajane** [comment]
I don't have an ETA yet, the remaining APIs to support before release are both fairly large so I don't want to make promises. Also it has to be an all-at-once release because it's a very breaking change from the current version of the library. If you can try the branch and let us know what you think that would probably help? It's subject to minor changes however!

**Comment by seopei** [comment]
Is the verify API fully implemented and well tested? Because we are just using this one. At the moment I am not so happy with the current implementation of the error handling at the master branch.

**Comment by lornajane** [comment]
@seopei ah, the Verify API part is pretty much done so I created a tag https://github.com/nexmo-community/nexmo-go/releases/tag/v0.12.0-alpha.1 if you want to try out this version and see if it's a better fit. The error handling (and accuracy with the API structs) is a major focus of this major change.

File: verify/api_default.go [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+	if err != nil {
+		return localVarReturnValue, localVarHTTPResponse, err
+	}
+
+	if localVarHTTPResponse.StatusCode >= 300 {
+		newErr := GenericOpenAPIError{
+			body:  localVarBody,
+			error: localVarHTTPResponse.Status,
+		}
+		if localVarHTTPResponse.StatusCode == 200 {
```
[the remark this answers is not in the document]
Comment (lornajane) [reply verify/api_default.go:139]: I'm not sure what to say about this! Everything in `verify/` is generated by [OpenAPI Generator](https://github.com/OpenAPITools/openapi-generator) - the aim is to have accurate structs and to be able to keep evolving the API functionality as things get added to the APIs. I did add a small edit to the generated code (to handle the Verify API rather legacy approach to errors) but my general aim is that this shouldn't be needed, and that the code could be regenerated to keep up with API changes. I'm not sure if you have a question or if I'm answering it!

[the remark this answers is not in the document]
Comment (seopei) [reply verify/api_default.go:139]: Ok, I see. It's generated code. I was just browsing your code and found this block of dead code so I left a review.

**Review by superdiana — APPROVED** [review]
LGTM

**Comment by superdiana** [comment]
I took a look at the conflicts and no major comments on those. (y)

**Pull request merged**